config/platforms/platform_intelsgx.h.in

```diff
@@ -0,0 +1,80 @@
+#include <endian.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <mbusafecrt.h>
+
+/*
+ * Intel SGX enclaves are completely isolated from the underlying
+ * kernel and therefore syscalls are not allowed by default. This
+ * means that getting the current time and locale information is
+ * impossible unless there is intervention by the enclave developers.
+ *
+ * Thus enclave developers must define DUK_USE_DATE_GET_NOW and
+ * DUK_USE_DATE_GET_LOCAL_TZOFFSET for their enclave either manually
+ * in duk_config.h or via other means (other config header, compiler
+ * flags). If these are not defined compilation will fail with a
+ * preprocessor error.
+ *
+ * DUK_USE_DATE_GET_NOW should return a duk_double_t value containing
+ * the current ECMASCript time.
+ * DUK_USE_DATE_GET_LOCAL_TZOFFSET should return a duk_int_t value
+ * containing the offset from UTC in seconds of the current time.
+ * See the datetime.rst file for more information, as well as the
+ * duk_bi_date.c, duk_bi_date_unix.c files for inspiration.
+ *
+ * DUK_USE_DATE_FORMAT_STRING and DUK_USE_DATE_PARSE_STRING macros are
+ * not mandatory and are therefore not defined here, but enclave
+ * developers are free to define them via duk_config.h or other means
+ * if they are implementing platform or locale scpecific date parsing
+ * and formatting.
+ *
+ * All related Duktape provided implementations regarding date and
+ * time are intentionally disabled.
+ *
+ * Similarly, any file I/O APIs are disabled since file I/O is not
+ * available in the enclave.
```

Comment:
There should no longer be any file I/O APIs in Duktape 2.x. Duktape 1.x had them but it proved to be more of a portability issue than expected.